include/libcyphal/transport/can/transport.hpp

```diff
@@ -133,40 +143,75 @@ class TransportImpl final : public ICanTransport
             return;
         }
 
-        const auto uint_ptr = reinterpret_cast<std::uintptr_t>(pointer);
-        CETL_DEBUG_ASSERT(uint_ptr > CanardMemory::DataOffset, "Invalid too small pointer.");
-        const auto canard_memory = reinterpret_cast<CanardMemory::Layout*>(uint_ptr - CanardMemory::DataOffset);
+        auto canard_memory = static_cast<CanardMemory*>(pointer);
+        --canard_memory;
 
         auto& self = getSelfFrom(ins);
         self.memory_.deallocate(canard_memory, canard_memory->size);
     }
 
+    template <typename Action>
+    void forEachMedia(Action action)
+    {
+        for (const auto media : media_array_)
+        {
+            CETL_DEBUG_ASSERT(media != nullptr, "Expected media interface.");
+            action(std::ref(*media));
+        }
+    }
+
+    template <typename T, typename Reducer>
+    T reduceMediaInto(T&& init, Reducer reducer) const
+    {
+        for (const auto media : media_array_)
+        {
+            CETL_DEBUG_ASSERT(media != nullptr, "Expected media interface.");
+            reducer(std::forward<T>(init), std::ref(*media));
```

Comment:
It is an interesting choice that the reducer modifies the argument in-place. Conventionally (including std::accumulate and std::reduce) the function returns the new value. std::for_each is different but its main purpose is to invoke side effects so it makes sense there.

Reply:
There actually exactly two (non-exclusive) approaches:
1. "conventional" reduce - return new value
2. and reduce "into".

You normally you use #1 for simple primitive types; and #2 for more complex/expensive stuff (like collection f.e.).
Look like currently my case is #1, but I wanted to make `reduceMediaInto` a bit more universal. So, if don't mind, I will keep it like it's currently, but will keep eye on this and refactor it (especially if there will be no #2 usages lately). ok?